### DISCUSSION TITLE
Array targeted new syntax

### DISCUSSION THREAD

**Discussion by NN---** [dotnet/csharplang#6058] [Language Ideas] [answered]
This syntax is not allowed.

```cs
class A { public int I; }

A[] a = new[] { new{ I = 1 } };
```

One must put a type inside the array:
```cs
class A { public int I; }

A[] a = new[] { new A { I = 1 } };
```

Although it is not required when it is not an array:

```cs
class A { public int I; }
class B { public A A; public B BB; }

// No problems
B b = new()
{
    A = new() { I = 1 },
    BB = new() { A = new() { I = 2 } }
};
```

**Comment by svick** [comment, marked as answer]
Assuming the syntax you're looking for is `A[] a = new[] { new() { I = 1 } };`, then this is a duplicate of https://github.com/dotnet/csharplang/issues/2701.

Also note that the less well-known syntax `A[] a = { new() { I = 1 } };` does work.
